insalan/payment/views.py

```diff
@@ -0,0 +1,160 @@
+        code = request.query_params.get("code")
+
+        if None in [trans_id, checkout_id, code]:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        transaction_obj = Transaction.objects.filter(payment_status=TransactionStatus.PENDING, id=trans_id, intent_id=checkout_id)
+        if len(transaction_obj) == 0:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        transaction_obj = transaction_obj[0]
+
+        if code != "success":
+            transaction_obj.payment_status = TransactionStatus.FAILED
+            transaction_obj.touch()
+            transaction_obj.save()
+
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        transaction_obj.payment_status = TransactionStatus.SUCCEEDED
+        transaction_obj.touch()
+        transaction_obj.save()
+
+        # Execute hooks
+        transaction_obj.run_success_hooks()
+
+        return Response(status=status.HTTP_200_OK)
+
+class ErrorView(generics.ListAPIView):
+    pass
+
+
+class PayView(generics.CreateAPIView):
+    permission_classes = [permissions.IsAuthenticated]
+    authentication_classes = [SessionAuthentication]
+    queryset = Transaction.objects.all()
+    serializer_class = serializers.TransactionSerializer
+
+    def create(self, request):
+        token = Token()
+        payer = request.user
+        data = request.data.copy()
+        data["payer"] = payer.id
+        logger.debug(f"data in view = {data}")  # contient des données
+        transaction = serializers.TransactionSerializer(data=data)
+        transaction.is_valid()
+        logger.debug(transaction.validated_data)
+        if transaction.is_valid(raise_exception=True):
+            transaction_obj = transaction.save()
+            # helloasso intent
+            helloasso_amount = int(
+                transaction_obj.amount * 100
+            )  # helloasso reads prices in cents
+            HELLOASSO_URL = getenv("HELLOASSO_ENDPOINT")
+            intent_body = {
+                "totalAmount": helloasso_amount,
+                "initialAmount": helloasso_amount,
+                "itemName": str(transaction_obj.id),
+                "backUrl": f"{getenv('HELLOASSO_BACK_URL')}?id={transaction_obj.id}",
+                "errorUrl": f"{getenv('HELLOASSO_ERROR_URL')}?id={transaction_obj.id}",
+                "returnUrl": f"{getenv('HELLOASSO_RETURN_URL')}?id={transaction_obj.id}",
+                "containsDonation": False,
+                "payer": {
+                    "firstName": payer.first_name,
+                    "lastName": payer.last_name,
+                    "email": payer.email,
+                },
+            }
+            headers = {
+                "authorization": "Bearer " + token.get_token(),
+                "Content-Type": "application/json",
+            }
+
+            checkout_init = requests.post(
+                f"{HELLOASSO_URL}/v5/organizations/insalan-test/checkout-intents",
```

Comment:
hardcoded organization here

Reply:
You know what I'm tired of all of these environment variables everywhere

time to make some configuration changes

Reply:
environment variables probably are the easiest way to ensure frontend and backend have a consistent configuration while not being accidentally committed, but I agree that all backend-only configuration should go elsewhere

Reply:
I moved all readings of variables into constants in the configuration, and added the organization slug name to those variables both in the infrastructure, and the backend